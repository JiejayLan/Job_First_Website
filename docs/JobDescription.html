<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: JobDescription</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: JobDescription</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>JobDescription<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description"><p>The class JobDescription will render the page of the job that user is interested
It will render the job information tab (data from AWS dynamoDB)
It will render the company information tab (data from AWS dynamoDB)
It will render the location tab (data from AWS dynamoDB with google map API)
It will render the applicant list only for the employer who posted the job</p></div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="JobDescription"><span class="type-signature"></span>new JobDescription<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobDescription.js.html">pages/jobDescription.js</a>, <a href="pages_jobDescription.js.html#line27">line 27</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="applyJob"><span class="type-signature"></span>applyJob<span class="type-signature"></span></h4>




<div class="description">
    <p>Update employee's applied job data
Condtion that check if the user applied to the job alreadu:
yes: don't update the applied job data
no: update the applied job data
Both conditions allow user to choose either stay at the page or go to profile page</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobDescription.js.html">pages/jobDescription.js</a>, <a href="pages_jobDescription.js.html#line218">line 218</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="componentDidMount"><span class="type-signature"></span>componentDidMount<span class="type-signature"></span></h4>




<div class="description">
    <p>Fetch all job information data from AWS by using job id
Update the job inforamtion by using job id
Check if the user is employer or employee:
if user is the company that posted the job:
yes: display the applicant list tab(with applicants' info) and hide the apply now button
no: just hide the apply now button
if user is employee: only display the aaply now button</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobDescription.js.html">pages/jobDescription.js</a>, <a href="pages_jobDescription.js.html#line67">line 67</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="loadingStatus"><span class="type-signature"></span>loadingStatus<span class="type-signature"></span></h4>










<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobDescription.js.html">pages/jobDescription.js</a>, <a href="pages_jobDescription.js.html#line298">line 298</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="state"><span class="type-signature"></span>state<span class="type-signature"> :Object</span></h4>




<div class="description">
    <p>the state object to store all data to display on the page</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type">Object</span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobDescription.js.html">pages/jobDescription.js</a>, <a href="pages_jobDescription.js.html#line33">line 33</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="visibleStatus"><span class="type-signature"></span>visibleStatus<span class="type-signature"></span></h4>










<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobDescription.js.html">pages/jobDescription.js</a>, <a href="pages_jobDescription.js.html#line309">line 309</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Begin to render and populated the page</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobDescription.js.html">pages/jobDescription.js</a>, <a href="pages_jobDescription.js.html#line318">line 318</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="AddEduForm.html">AddEduForm</a></li><li><a href="AddExpForm.html">AddExpForm</a></li><li><a href="businessProfile.html">businessProfile</a></li><li><a href="Contact.html">Contact</a></li><li><a href="CreateAddressForm.html">CreateAddressForm</a></li><li><a href="JobDescription.html">JobDescription</a></li><li><a href="JobList.html">JobList</a></li><li><a href="ModalForm.html">ModalForm</a></li><li><a href="PostJobForm.html">PostJobForm</a></li><li><a href="Profile.html">Profile</a></li><li><a href="UpdateAddressForm.html">UpdateAddressForm</a></li></ul><h3>Global</h3><ul><li><a href="global.html#AppliedJob">AppliedJob</a></li><li><a href="global.html#disabledDate">disabledDate</a></li><li><a href="global.html#Education">Education</a></li><li><a href="global.html#Experiences">Experiences</a></li><li><a href="global.html#formatDate">formatDate</a></li><li><a href="global.html#General">General</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.2</a> on Fri May 24 2019 10:55:13 GMT-0400 (Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>