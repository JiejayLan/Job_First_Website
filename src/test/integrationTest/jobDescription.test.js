import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import JobDescription from '../../pages/jobDescription';

describe('Job Description screenshot test', () => {
    test('Job Description page should render correctly', () => {
        const renderer = new ReactShallowRenderer();
        renderer.render(<JobDescription />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
});