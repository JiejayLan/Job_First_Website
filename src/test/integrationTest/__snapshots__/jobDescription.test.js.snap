// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Description screenshot test Job Description page should render correctly 1`] = `
<div>
  <Spin
    size="default"
    spinning={false}
    tip="Please wait for a moment"
    wrapperClassName=""
  >
    <h2
      style={
        Object {
          "margin": "10px 0",
        }
      }
    >
      
    </h2>
    <Popover
      mouseEnterDelay={0.1}
      mouseLeaveDelay={0.1}
      overlayStyle={Object {}}
      placement="top"
      transitionName="zoom-big"
      trigger="hover"
    >
      <div>
         - 
      </div>
    </Popover>
    <Popover
      content="We will use your default information to apply to the job"
      mouseEnterDelay={0.1}
      mouseLeaveDelay={0.1}
      overlayStyle={Object {}}
      placement="top"
      transitionName="zoom-big"
      trigger="hover"
    >
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        Apply Now
      </Button>
    </Popover>
    <PopOutWindow
      cancelText="Stay here"
      content="Thanks for applying to this job, you will be heard back from the employer shortly."
      isLoading={[Function]}
      isVisible={[Function]}
      okText="Go to profile page"
      show={false}
      userId=""
    />
    <Tabs
      defaultActiveKey="1"
      hideAdd={false}
      tabPosition="top"
    >
      <TabPane
        placeholder={null}
        tab="Job"
      >
        <div>
          <jobDetails
            jobInfo={
              Object {
                "clickedCount": 0,
                "description": "",
                "requirements": Array [],
                "title": "",
                "type": "",
              }
            }
          />
        </div>
      </TabPane>
      <TabPane
        placeholder={null}
        tab="Company"
      >
        <div>
          <companyDetail
            companyInfo={Object {}}
          />
        </div>
      </TabPane>
      <TabPane
        placeholder={null}
        tab="Location"
      >
        <div>
          <location
            locationInfo={Object {}}
          />
        </div>
      </TabPane>
    </Tabs>
  </Spin>
</div>
`;
